l10n_it_fatturapa_out_sale/wizard/wizard_export_fatturapa.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl).
+
+from openerp import models
+
+
+class WizardExportFatturapa(models.TransientModel):
+    _inherit = "wizard.export.fatturapa"
+
+    def setFatturaElettronicaBody(self, invoice, body):
+        rel_docs_model = self.env['fatturapa.related_document_type']
+        if invoice.partner_id.fatturapa_sale_order_data:
+            # if sale_order refer to the whole invoice create only 1 rel doc
+            if invoice.picking_ids and set(self.env['sale.order'].search([
+                ('name', 'in', invoice.picking_ids.mapped('origin'))]).mapped(
+                'client_order_ref')) == 1 or \
+                    len(set(invoice.invoice_line.mapped('origin'))) == 1:
+                # check if already present ref for this invoice
+                existing_rel_docs = rel_docs_model.search([
+                    ('invoice_id', '=', invoice.id),
+                    ('type', '=', 'order')
+                ])
+                if existing_rel_docs:
+                    existing_rel_docs.unlink()
```

Comment:
In generale mi preoccupa un po' eliminare dati silenziosamente. L'utente potrebbe aver inserito manualmente qualcosa, che poi viene eliminato.

Non si potrebbe aggiungere il documento correlato direttamente nell'XML, senza scriverlo nel DB?

Reply:
mmm, anche, ci ragiono

Reply:
+1

Reply:
@scigghia @eLBati ho modificato il modulo per generare l'xml direttamente, ho aggiunto anche il flag direttamente in fattura per l'opzione